linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/Emitter.scala

```diff
@@ -74,12 +75,14 @@ final class Emitter(config: Emitter.Config) {
 
   val injectedIRFiles: Seq[IRFile] = PrivateLibHolder.files
 
-  def emit(unit: LinkingUnit, logger: Logger): Result = {
-    val WithGlobals(body, globalRefs) = emitInternal(unit, logger)
+  def emit(moduleSet: ModuleSet, logger: Logger): Result = {
+    val WithGlobals(body, globalRefs) = emitInternal(moduleSet, logger)
 
     moduleKind match {
       case ModuleKind.NoModule =>
-        val topLevelVars = unit.topLevelExports.map(_.exportName)
+        val topLevelVars = moduleSet.modules
```

Comment:
No, we cannot. We can have 0 or 1 module here. Not sure how to improve this though.

Reply:
Perhaps just `assert(moduleSet.modules.size <= 1)` and then get `moduleSet.modules.head`?

It may be less "type-safe" but it would be clear what's happening.